src/CMakeLists.txt

```diff
@@ -339,8 +339,24 @@ add_subdirectory(box)
 include_directories(${EXTRA_BOX_INCLUDE_DIRS})
 
 # Save CMAKE_XXX_FLAGS from this directory for config.h (used in --version)
-set(TARANTOOL_C_FLAGS ${CMAKE_C_FLAGS} PARENT_SCOPE)
-set(TARANTOOL_CXX_FLAGS ${CMAKE_CXX_FLAGS} PARENT_SCOPE)
+string(REPLACE ";" " " TARANTOOL_CFLAGS_LTO "${CFLAGS_LTO}")
+
+set(TARANTOOL_BUILD_TYPES Debug Release RelWithDebInfo MinSizeRel)
+list(FIND TARANTOOL_BUILD_TYPES "${CMAKE_BUILD_TYPE}" TARANTOOL_BUILD_TYPE)
+
+if(${TARANTOOL_BUILD_TYPE} EQUAL -1)
+    message(FATAL_ERROR "Unknown config passed: ${CMAKE_BUILD_TYPE}")
```

Comment:
@Gumix, regarding this error: to me it's not good to silently accept misspelling a user has introduced, but I'm not sure if it's OK for Tarantool too. Awaiting for @locker's opinion.